diff --git a/dist/base.d.ts b/dist/base.d.ts
index 2231347d7fcf133e1cdee1f4ce2a49658b743866..f0100bf6c40107b623e05e760e04dca1a0eb16a1 100644
--- a/dist/base.d.ts
+++ b/dist/base.d.ts
@@ -2,20 +2,20 @@ export type Meta = ReadonlyMap<string, unknown>;
 /**
  * The basic structure that contains
  */
-export interface TypedMessage {
+export interface TypedMessage<K extends string = string, V extends unknown = unknown> {
     /**
      * The metadata this message contains.
      * Not expected to display to normal users.
      *
      * It will be serialized to a JSON object.
      */
-    readonly meta?: ReadonlyMap<string, unknown>;
+    readonly meta?: ReadonlyMap<K, V>
     /**
      * The type of this message.
      *
      * Custom message must starts with 'x-' (means extension).
      */
-    readonly type: string;
+    readonly type: string
 }
 export type SerializableTypedMessages = SerializableTypedMessage<number> | NonSerializableWithAltTypedMessage;
 /**
diff --git a/src/base.ts b/src/base.ts
index 5167aba1987088e636a4b95cdf176cbfbd961118..58fd52017bec1b03df39679628230985d60316d6 100644
--- a/src/base.ts
+++ b/src/base.ts
@@ -2,14 +2,14 @@ export type Meta = ReadonlyMap<string, unknown>
 /**
  * The basic structure that contains
  */
-export interface TypedMessage {
+export interface TypedMessage<K extends string = string, V extends unknown = unknown> {
     /**
      * The metadata this message contains.
      * Not expected to display to normal users.
      *
      * It will be serialized to a JSON object.
      */
-    readonly meta?: ReadonlyMap<string, unknown>
+    readonly meta?: ReadonlyMap<K, V>
     /**
      * The type of this message.
      *
