diff --git a/dist/esm/exports/index.js b/dist/esm/exports/index.js
index 7b0c23279a88713a061f3f14fa3d544f53e5eabd..881b56f6d7029f1286459aaa7c1240b1d502ed1a 100644
--- a/dist/esm/exports/index.js
+++ b/dist/esm/exports/index.js
@@ -2,13 +2,21 @@ import { CoreHelperUtil } from '@reown/appkit-controllers';
 import { AppKit } from '../src/client/appkit.js';
 import { PACKAGE_VERSION } from './constants.js';
 // -- Utils & Other -----------------------------------------------------
+export {
+    AccountController,
+    ApiController as CoreApiController,
+    AssetController as CoreAssetController,
+    AssetUtil as CoreAssetUtil,
+    ChainController as CoreChainController,
+    ConnectionController as CoreConnectionController,
+    ConnectorController as CoreConnectorController, CoreHelperUtil, OptionsController as CoreOptionsController, RouterController as CoreRouterController, StorageUtil as CoreStorageUtil
+} from '@reown/appkit-controllers';
 export * from '../src/utils/index.js';
-export { CoreHelperUtil, AccountController } from '@reown/appkit-controllers';
+export { AppKit };
 export function createAppKit(options) {
     return new AppKit({
         ...options,
         sdkVersion: CoreHelperUtil.generateSdkVersion(options.adapters ?? [], 'html', PACKAGE_VERSION)
     });
 }
-export { AppKit };
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/dist/types/exports/index.d.ts b/dist/types/exports/index.d.ts
index 068209f6b86260c838b0fe04f043e2e9d5e526a0..4edbe93d73a21bf0f6ce5ea4c7e84c656d4bc2c5 100644
--- a/dist/types/exports/index.d.ts
+++ b/dist/types/exports/index.d.ts
@@ -1,10 +1,19 @@
 import { AppKit } from '../src/client/appkit.js';
 import type { AppKitOptions } from '../src/utils/TypesUtil.js';
-export * from '../src/utils/index.js';
+export type { CaipAddress, CaipNetwork, CaipNetworkId } from '@reown/appkit-common';
 export type * from '@reown/appkit-controllers';
-export type { CaipNetwork, CaipAddress, CaipNetworkId } from '@reown/appkit-common';
-export { CoreHelperUtil, AccountController } from '@reown/appkit-controllers';
-export type CreateAppKit = Omit<AppKitOptions, 'sdkType' | 'sdkVersion' | 'basic'>;
-export declare function createAppKit(options: CreateAppKit): AppKit;
+export {
+    AccountController,
+    ApiController as CoreApiController,
+    AssetController as CoreAssetController,
+    AssetUtil as CoreAssetUtil,
+    ChainController as CoreChainController,
+    ConnectionController as CoreConnectionController,
+    ConnectorController as CoreConnectorController, CoreHelperUtil, OptionsController as CoreOptionsController, RouterController as CoreRouterController, StorageUtil as CoreStorageUtil
+} from '@reown/appkit-controllers';
+export * from '../src/utils/index.js';
 export { AppKit };
 export type { AppKitOptions };
+export type CreateAppKit = Omit<AppKitOptions, 'sdkType' | 'sdkVersion' | 'basic'>;
+export declare function createAppKit(options: CreateAppKit): AppKit;
+
