diff --git a/esm/index.js b/esm/index.js
index fcc5f5c3112bba4002eff578edaa4d8576ba41f0..a218f8012f2dca0dd0453666d123dab51e8042ab 100644
--- a/esm/index.js
+++ b/esm/index.js
@@ -16,10 +16,6 @@ export { HttpContext, httpContextIntegration } from './integrations/httpcontext.
 export { LinkedErrors, linkedErrorsIntegration } from './integrations/linkederrors.js';
 export { TryCatch, browserApiErrorsIntegration } from './integrations/trycatch.js';
 import * as index from './integrations/index.js';
-export { InternalReplay as Replay, internalGetReplay as getReplay, internalReplayIntegration as replayIntegration } from '@sentry/replay';
-export { ReplayCanvas, replayCanvasIntegration } from '@sentry-internal/replay-canvas';
-export { Feedback, feedbackIntegration, sendFeedback } from '@sentry-internal/feedback';
-export { captureConsoleIntegration, contextLinesIntegration, debugIntegration, dedupeIntegration, extraErrorDataIntegration, httpClientIntegration, reportingObserverIntegration, rewriteFramesIntegration, sessionTimingIntegration } from '@sentry/integrations';
 export { BrowserTracing, browserTracingIntegration, defaultRequestInstrumentationOptions, instrumentOutgoingRequests, startBrowserTracingNavigationSpan, startBrowserTracingPageLoadSpan } from '@sentry-internal/tracing';
 export { makeBrowserOfflineTransport } from './transports/offline.js';
 export { onProfilingStartRouteTransaction } from './profiling/hubextensions.js';
