diff --git a/dist/tippy-react.esm.js b/dist/tippy-react.esm.js
index 8fc6085b0e55b80757fa2eee59bb830947c624a5..2b2bf90c0648ff1035f399761116cc7f7a58408d 100644
--- a/dist/tippy-react.esm.js
+++ b/dist/tippy-react.esm.js
@@ -357,7 +357,7 @@ function TippyGenerator(tippy) {
     return /*#__PURE__*/React.createElement(React.Fragment, null, children ? /*#__PURE__*/cloneElement(children, {
       ref: function ref(node) {
         mutableBox.ref = node;
-        preserveRef(children.ref, node);
+        preserveRef(children?.props?.ref, node);
       }
     }) : null, mounted && /*#__PURE__*/createPortal(render ? render(toDataAttributes(attrs), singletonContent, mutableBox.instance) : content, mutableBox.container));
   }
@@ -514,7 +514,7 @@ var forwardRef = (function (Tippy, defaultProps) {
       React.createElement(Tippy, Object.assign({}, defaultProps, props), children ? /*#__PURE__*/cloneElement(children, {
         ref: function ref(node) {
           preserveRef(_ref2, node);
-          preserveRef(children.ref, node);
+          preserveRef(children?.props?.ref, node);
         }
       }) : null)
     );
