diff --git a/dist/index.cjs b/dist/index.cjs
index 62cd951e575ea98c41696c76b0049fc35bab24ba..a444b9202d78bc3d9c948b90f6ad7f22fda92e5e 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -2,7 +2,8 @@
 
 var path2 = require('path');
 var module$1 = require('module');
-var defineEnvPlugin_js = require('next/dist/build/webpack/plugins/define-env-plugin.js');
+// https://github.com/storybookjs/vite-plugin-storybook-nextjs/issues/39
+var defineEnvPlugin_js = require('next/dist/build/define-env.js');
 var fs3 = require('fs');
 var nextEnv = require('@next/env');
 var Log = require('next/dist/build/output/log.js');
diff --git a/dist/index.js b/dist/index.js
index 364186ae63bbc0d92bb1539cb2a79dee947c40a0..e5e5684624b968ce8f5ab105a11e0662289090eb 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1,6 +1,7 @@
 import path2, { resolve, join, sep } from 'node:path';
 import { createRequire } from 'node:module';
-import { getDefineEnv } from 'next/dist/build/webpack/plugins/define-env-plugin.js';
+// https://github.com/storybookjs/vite-plugin-storybook-nextjs/issues/39
+import { getDefineEnv } from 'next/dist/build/define-env.js';
 import fs3 from 'node:fs';
 import * as nextEnv from '@next/env';
 import Log from 'next/dist/build/output/log.js';
